clocksilencer.ino

```diff
@@ -20,12 +21,16 @@ void setup() {
   Serial.begin(115200);
   hwSerial.begin(9600, SERIAL_8N1, 16,
                  17);  // second serial port to serve mp3 player
-  //if (rtcTimeWasAlreadySetFromNTP == false) {
-    synchroniseTime();
-  //} else {
-  //  setenv("TZ", "CET-1CEST,M3.5.0,M10.5.0/3", 3);  // repair timezone
-  // tzset();
-  //}
+  if (rtcTimeWasAlreadySetFromNTP == false) {
+    synchroniseExternalTime(); //but here we synchronise time from ntp and send it to both external and internal rtc
```

Comment:
Why synchronize to both: external and internal?

It should be enough to set external RTC only and always read from there.

I mean:

```c++
if (rtcTimeWasAlreadySetFromNTP == false) {
    synchroniseExternalTime();
} else {
    // no need to do anything: external clock is able to provide real-time
}
```

Not sure how to read Polish time from RTC, but  we could set the time inepoch seconds (UTC) converting them to PL every time we wake up from deep sleep (to make sure we ring correct number after time change between summer and winter time).

- https://github.com/The-Coobaz/crazyclock/blob/main/crazyclock.ino#LL71C21-L71C34

Reply:
You are right about the synchronisation. Since we have a reliable time source onboard we can use it anytime.
But I am not concerned about Polish time. In my opinion there's no need to convert time to epoch. Maybe it's just enough to use `timezone.h` library?

Reply:
No. Only on first Boot we update time from utp and put into external rtc.
Then, on every next Boot we use external rtc to update the internal one.

Reply:
Yes, this is also how I understand it. But I don't think we need to update the internal one.

We could just ignore it. And work **only with external** RTC.

I think this would be simpler to understand.

I other words: because of some troubles with internal clock during deep sleep we can not use it at all and rely only on external RTC (powered by a battery).

Reply:
So you suggest to use only Timer method and rely completely on external rtc, am I right?

Reply:
yes, that is my suggestion :) what do you think about it?

maybe are there are some disadvantages of such approach?

Reply:
I believe this may simplify the code indeed :)